python/vulnerabilities/zero_day_vuln_search.py

```diff
@@ -0,0 +1,122 @@
+import os
+import sys
+import json
+import requests
+
+VERSION = "1.0.0"
+
+# Process an HTTP error by printing and log.error
+def process_http_error(msg, response, url):
+    if response.text is None:
+        print(f"{msg} HTTP Error: {response.status_code} with {url}")
+    else:
+        print(f"{msg}, {url} status_code: {response.status_code} info: {response.text}")
+
+# Performs a search for vulnerabilities with zero day information.
+def search_vulns_for_zero_day(base_url, headers):
+    search_vulns_url = f"{base_url}/vulnerabilities/search"
+
+    query_params = "?zero_day[]=true&fields=id,created_at,identifiers,last_seen_time,cve_id,description"
+    search_vulns_url += query_params
+
+    response = requests.get(search_vulns_url, headers=headers)
+    if response.status_code != 200:
+        process_http_error(f"Vulnerability Search API Error", response, search_vulns_url)
+        sys.exit(1)
+
+    return response.json()
+
+# Obtains the Talos zero day data.
+def get_vuln_data(base_url, headers, vuln_id):
+    show_vuln_url = f"{base_url}/vulnerabilities/{vuln_id}"
+
+    response = requests.get(show_vuln_url, headers=headers)
+    if response.status_code != 200:
+        process_http_error(f"Show Vulnerability API Error", response, show_vuln_url)
+        sys.exit(1)
+
+    vuln_resp = response.json()
+    return vuln_resp['vulnerability']
+
+# Checks if a Talos report URL exists.
+def talos_url_exists(talos_id):
+    talos_vuln_url = f"https://www.talosintelligence.com/vulnerability_reports/{talos_id}"
+
+    try:
+        response = requests.get(talos_vuln_url)
```

Comment:
should we not be logging talos url does not exist for this talos_id,

Not sure how talos works but doesn't this require API key or some sort of authentication ?

Reply:
Good point.  There is no authentication.  Adding logging is good point.  I can go through and add it.